akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/AdaptedAskSpec.scala

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (C) 2009-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.actor.typed.scaladsl
+
+import akka.actor.typed.{ ActorRef, TypedAkkaSpec, TypedAkkaSpecWithShutdown }
+import akka.testkit.typed.TestKit
+import akka.testkit.typed.scaladsl.TestProbe
+import akka.util.Timeout
+
+import scala.concurrent.duration._
+import scala.util.{ Failure, Success }
+
+class AdaptedAskSpec extends TestKit() with TypedAkkaSpecWithShutdown {
+
+  sealed trait Protocol
+
+  sealed trait OtherActorProtocol
+  case object OtherActorPong
+  case class OtherActorPing(respondTo: ActorRef[OtherActorPong.type]) extends OtherActorProtocol
+
+  sealed trait ActorProtocol
+  case class TriggerPing(id: Long) extends ActorProtocol
+  case class GotPong(id: Long) extends ActorProtocol
+  case class PongTimedOut(id: Long) extends ActorProtocol
+
+  "Asking another actor" should {
+
+    "work just fine" in {
+
+      val otherActor = spawn(Actor.immutable[OtherActorProtocol] { (ctx, msg) ⇒
+        msg match {
+          case OtherActorPing(respondTo) ⇒
+            respondTo ! OtherActorPong
+            Actor.same
+        }
+      })
+
+      val probe = TestProbe[AnyRef]()
+
+      implicit val timeout = Timeout(1.second)
+      val actor = spawn(Actor.immutable[ActorProtocol] { (ctx, msg) ⇒
+
+        msg match {
+          case TriggerPing(id) ⇒
+            ctx.ask(otherActor, OtherActorPing) {
+              case Success(OtherActorPong) ⇒ GotPong(id)
+              case Failure(_)              ⇒ PongTimedOut(id)
+            }
```

Comment:
hm, then we have two ways of doing `ask`, one via `ctx` and one via `ActorRef`. For Scala we could perhaps make this an implicit so that it still looks like `otherActor.ask`?

Reply:
I think the `AskPattern` way of asking will be used very seldom inside of actors so that this will effectively be the way (tm) to do ask. 

The scenarios where you'd want to use `AskPattern` directly that I can think of is for asking outside of actors, or pipe to other-actor.

Since `ActorContext` is a parameter to the lambda that defines the actor behavior a user would have to mark the entire parameter list implicit, or capture the context in one additional implicit value to do the `ActorRef.ask`, seems like a lot of boilerplate for relatively little convenience to me?

Reply:
I am also not sure if the following use of ask deserves an abstraction:
```scala
import akka.actor.typed.scaladsl.AskPattern._
import ctx.executionContext

otherActor ? OtherActorPing onComplete {
  case Success(OtherActorPong) ⇒ ctx.self ! GotPong(id)
  case Failure(_)              ⇒ ctx.self ! PongTimedOut(id)
}
```

Reply:
Hmm, fair point.

Reply:
Given that I can't think of a shorter way to express the same thing, or things we can leave out, and that just doing the same thing takes roughly the same amount of effort/code I think we should can this PR.

Reply:
I really think while @2m's proposal is true, it's not very helpful to people new to Akka.
And even just existing people new to Typed. It's such a simple quality-of-life improvement that we should add it, even if there's a close but not that easy to discover way to do the same thing. 

With such "quality of life" API improvements we can get away from "akka is so hard, omg!"
This API can also work without having people to import implicits like the dispatcher, since it could use the actor's default one, which should be fine.

With the implicit it may be hard perhaps, or ask needs a redesign perhaps, since in the existing form it'd end up being ambigious I think? (since one ask is (ask(ref => ...) and this one is `ask(msg =>` which is not clear which is which. 

Perhaps this one is `askAdapt`? Or we need ask to be a builder that returns an `Ask` that has methods like `.pipeTo` etc